### PR TITLE
chore: remove cluster driver registrar

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
According to Kubernetes CSI Developer Documentation, the cluster-driver-registrar side-car which was used to create CSIDriver objects in Kubernetes 1.13 has been deprecated for Kubernetes 1.16. No cluster-driver-registrar has been released for Kubernetes 1.14 and later.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
remove cluster driver registrar
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
could you make sure this PR works in k8s 1.14? There are quite a lot users are using 1.14 now, we need to make sure it works. thanks  @ZeroMagic

File: deploy/csi-azurefile-driver.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: storage.k8s.io/v1beta1
+kind: CSIDriver
+metadata:
+  name: file.csi.azure.com
+spec:
+  attachRequired: true
+  podInfoOnMount: true
+  volumeLifecycleModes:  # added in Kubernetes 1.16
```
Comment (andyzhangx) [inline deploy/csi-azurefile-driver.yaml:9]: pls remove this, there are users are still using 1.16, we don't want to break those users.

Comment (ZeroMagic) [reply deploy/csi-azurefile-driver.yaml]: done

**Pull request merged**